plugins/ssh-plugin/src/util/ssh-secret-helper.ts

```diff
@@ -0,0 +1,129 @@
+        CREDENTIALS_SECRET_NAME,
+        this.getWorkspaceNamespace()
+      );
+      data = request.body.data;
+    } catch (e) {
+      theia.window.showErrorMessage('Failed to retrieve the SSH secret' + e);
+    }
+    if (data && data[name]) {
+      const privateKey = Buffer.from(data[name], 'base64').toString();
+      const publicKey = Buffer.from(data[name + '.pub'], 'base64').toString();
+      return { name, publicKey, privateKey };
+    } else {
+      throw new Error(`SSH key with name ${name} was not found`);
+    }
+  }
+
+  async getAll(): Promise<SshPair[]> {
+    let data: { [key: string]: string } | undefined;
+    try {
+      const request = await this.getK8sCoreApi().readNamespacedSecret(
+        CREDENTIALS_SECRET_NAME,
+        this.getWorkspaceNamespace()
+      );
+      data = request.body.data;
+    } catch (e) {
+      theia.window.showErrorMessage('Failed to retrieve the SSH secret' + e);
+    }
+    const sshKeys: SshPair[] = [];
+    if (data) {
+      Object.keys(data)
+        .filter(key => key.endsWith('.pub'))
+        .forEach(key => {
+          const name = key.substring(0, key.length - 4);
+          sshKeys.push({
+            name,
+            publicKey: Buffer.from(data![key], 'base64').toString(),
+            privateKey: Buffer.from(data![name], 'base64').toString(),
+          });
+        });
+    }
+    return sshKeys;
+  }
+
+  async store(ssh: SshPair): Promise<void> {
+    const client = this.getK8sCoreApi();
+    client.defaultHeaders = {
+      Accept: 'application/json',
+      'Content-Type': k8s.PatchUtils.PATCH_FORMAT_STRATEGIC_MERGE_PATCH,
+    };
+    try {
+      await client.patchNamespacedSecret(CREDENTIALS_SECRET_NAME, this.getWorkspaceNamespace(), {
+        data: { [ssh.name]: Buffer.from(ssh.privateKey).toString('base64') },
+      });
+      await client.patchNamespacedSecret(CREDENTIALS_SECRET_NAME, this.getWorkspaceNamespace(), {
+        data: { [ssh.name + '.pub']: Buffer.from(ssh.publicKey).toString('base64') },
+      });
+    } catch (e) {
+      theia.window.showErrorMessage('Failed to store the SSH key pair' + e);
+    }
+  }
+
+  async delete(name: string): Promise<void> {
+    try {
+      const patchPrivate = [
+        {
+          op: 'remove',
+          path: `/data/${name}`,
+        },
+      ];
+      const patchPublic = [
+        {
+          op: 'remove',
+          path: `/data/${name}.pub`,
+        },
+      ];
+      const client = this.getK8sCoreApi();
+      client.defaultHeaders = { Accept: 'application/json', 'Content-Type': k8s.PatchUtils.PATCH_FORMAT_JSON_PATCH };
+      await client.patchNamespacedSecret(CREDENTIALS_SECRET_NAME, this.getWorkspaceNamespace(), patchPrivate);
+      await client.patchNamespacedSecret(CREDENTIALS_SECRET_NAME, this.getWorkspaceNamespace(), patchPublic);
+    } catch (e) {
+      theia.window.showErrorMessage('Failed to delete the SSH secret' + e);
+    }
+  }
+
+  private getWorkspaceNamespace(): string {
```

Comment:
I think you can use `getCurrentNamespace()` method from `workspace` namespace to get Namespace

https://github.com/eclipse-che/che-theia/blob/main/extensions/eclipse-che-theia-plugin/src/che-proposed.d.ts#L32

Reply:
done